common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -31,35 +33,51 @@
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public final class PlayerModel extends Model {
+    private static final String ATHENA_USER_NOT_FOUND = "User not found";
     private static final Pattern GHOST_WORLD_PATTERN = Pattern.compile("^_(\\d+)$");
+
+    // If there is a failure with the API, give it time to recover
+    private static final int ERROR_TIMEOUT_MINUTE = 5;
+
+    // Max amount of overall lookup errors
     private static final int MAX_ERRORS = 5;
 
+    // Max amount of errors for a single user,
+    // before disabling lookups for them
+    private static final int MAX_USER_ERRORS = 3;
+
     private final Map<UUID, WynntilsUser> users = new ConcurrentHashMap<>();
+    private final Set<UUID> usersWithoutWynntilsAccount = ConcurrentHashMap.newKeySet();
     private final Set<UUID> fetching = ConcurrentHashMap.newKeySet();
     private final Map<UUID, Integer> ghosts = new ConcurrentHashMap<>();
-    private int errorCount;
     private final Map<UUID, String> nameMap = new ConcurrentHashMap<>();
 
+    // The size of this set is the count of errors in the last ERROR_TIMEOUT_MINUTE minutes.
+    // This is used to avoid spamming the API.
+    private final TimedSet<Object> errors =
+            new TimedSet<>(ERROR_TIMEOUT_MINUTE, TimeUnit.MINUTES, true, ConcurrentHashMap::newKeySet);
+    private final Map<UUID, Integer> userFailures = new ConcurrentHashMap<>();
+
     public PlayerModel() {
         super(List.of());
-
-        errorCount = 0;
+        errors.clear();
+        userFailures.clear();
     }
 
     // Returns true if the player is on the same server and is not a npc
     public boolean isLocalPlayer(Player player) {
-        return !isNpc(player) && !(isPlayerGhost(player));
+        return !isNpc(player) && !isPlayerGhost(player);
     }
 
     public boolean isLocalPlayer(String name) {
         // Wynn uses TeamNames for player names that are online
-        return !isNpc(StyledText.fromString(name))
+        return !isNpcName(StyledText.fromString(name))
                 && McUtils.mc().level.getScoreboard().getTeamNames().contains(name);
     }
 
     public boolean isNpc(Player player) {
         StyledText scoreboardName = StyledText.fromString(player.getScoreboardName());
-        return isNpc(scoreboardName);
+        return isNpcName(scoreboardName) || isNpcUuid(player.getUUID());
```

Comment:
Is not isNpcUuid enough?

Reply:
After Wynn v2.1, it is. Not right now.

Reply:
I see. I've lost track of the Wynn versioning scheme.